CONTRIBUTING.md

```diff
@@ -16,7 +18,48 @@ First off all, thank you for taking the time to contribute!
 
 Help us keep exercism welcoming. Please read and abide by the [Code of Conduct][coc].
 
-## Setting up your local development environment
+## What is the walkthrough?
+
+### Background
+
+Looking at the data, we noticed that a significant number of users have never
+completed a single exercise on exercism. We believe this is because setting it up
+isn't easy. This walkthrough aims to solve that.
+
+To do exercises on exercism, installing a CLI is required. This walkthrough aims
+to guide users throughout the several installation steps.
+
+### Structure
+
+The walkthrough is strucured in the following way:
+
+1. The walkthrough is a step-by-step guide. This should prevent users from
+getting lost.
+1. At the end of each step, the user must be able to confirm whether they've
+done the step correctly. This should prevent users from installing incomplete
+packages which may lead to harder troubleshooting in the future.
+1. When the user encounters a problem during a step, the user should be given
+steps to self-diagnose or troubleshoot the problem on their own.
+
+## How can I help?
+
+Again, thank you for taking the time to contribute! You could contribute to
+the walkthrough by doing these:
+
+1. Giving feedback on the walkthrough. Answering questions such as,
+**How can we improve it?**, **What step was the most confusing?**,
+**What additional steps did you have to take?** would help us make the
+walkthrough cater to more people.
+1. Looking through the issues and submitting a patch. Upcoming features and
+bugs to fix would all be in this project's issue tracker.
+1. Testing out the walkthrough. As each user may have a unique setup, it'll
+be helpful to have a hand in testing out the walkthrough on different machines.
+1. Writing better instructions. If you have a better way to describe a step in
+mind, a PR submission would be greatly appcreiated.
```

Comment:
s/appcreiated/appreciated/

I wonder if this might make sense as the second step. That way it's

- test out the instructions to find gaps
- submit patches for any suggested improvements that you have
- [something, not sure exactly what the third bullet means]
- pick an issue out of the issue tracker

Reply:
Good point with ordering them with the easiest tasks first :+1: